manim/mobject/geometry.py

```diff
@@ -187,7 +212,9 @@ def reset_endpoints_based_on_tip(self, tip, at_start):
             self.put_start_and_end_on(self.get_start(), tip.base)
         return self
 
-    def asign_tip_attr(self, tip, at_start):
+    def asign_tip_attr(
+        self, tip: ArrowTriangleFilledTip, at_start: bool
+    ) -> Union[NumberLine, Arrow, Vector, DoubleArrow]:
```

Comment:
Stuff like this should probably have newtypes, or maybe a superclass.

Reply:
I would, but a pr is about to nuke all those classes lol

Reply:
Maybe this should be based on that change then